### PR TITLE
Upgrade to zipkin 3.14.0 and replace SpanCollector with Reporter

### PR DESCRIPTION
**Pull request by jplock**

### CODE REVIEW

**Comment by jplock** [comment]
@adriancole is this the correct path forward?  For the HTTP sender, do you recommend the OkHttp one or should the URLConnectionSender be sufficient?

File: zipkin-core/src/main/java/com/smoketurner/dropwizard/zipkin/KafkaZipkinFactory.java [hunk 1]
```diff
@@ -69,17 +70,17 @@ public void setTopic(String topic) {
...
+        final ReporterMetrics metricsHandler = new DropwizardReporterMetrics(
                 environment.metrics());
 
-        final KafkaSpanCollector.Config kafkaConfig = KafkaSpanCollector.Config
-                .builder(bootstrapServers).topic(topic).build();
-        final SpanCollector spanCollector = KafkaSpanCollector
-                .create(kafkaConfig, metricsHandler);
+        final KafkaSender sender = KafkaSender.builder()
+                .bootstrapServers(bootstrapServers).topic(topic).build();
+        final AsyncReporter<Span> reporter = AsyncReporter.builder(sender)
```
Comment (adriancole) [inline zipkin-core/src/main/java/com/smoketurner/dropwizard/zipkin/KafkaZipkinFactory.java]: note sender and also AsyncReporter are closeable (individually: close reporter before sender). also, if there's any "post construct" phase, you could call reporter.check() on it

Comment (jplock) [inline zipkin-core/src/main/java/com/smoketurner/dropwizard/zipkin/KafkaZipkinFactory.java]: I'll set up a ManagedSender instance so Dropwizard can shut it down when Jetty is stopped.  http://www.dropwizard.io/1.0.2/docs/manual/core.html#managed-objects

Thanks!

**Comment by jplock** [comment]
@adriancole I was just looking at the AsyncReporter, I'll set up a manager for that too.  Is it possible to update the zipkin version in the zipkin-reporters to 1.13.0?  That would fix my dependency convergence errors on this PR.

**Comment by jplock** [comment]
@adriancole also, should the sender's `close()` method be called when closing the reporter?  https://github.com/openzipkin/zipkin-reporter-java/blob/master/core/src/main/java/zipkin/reporter/AsyncReporter.java#L239

**Comment by adriancole** [comment]
sure I'll bump reporter now

**Comment by adriancole** [comment]
yep, close reporter, then sender

**Comment by adriancole** [comment]
0.6.1 is syncing to central now

**Pull request merged**